### ISSUE TITLE
[FP] Unused function result on fluent interface

### ISSUE THREAD

**Issue by funivan** [kalessil/phpinspectionsea#1158]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive                                                |
| Plugin         |  Php Inspections (EA Ultimate) v2.0.11 |
| Language level | PHP 7.1                        |

Create a test file 
```php
<?
  declare(strict_types=1);
  final class User {
    /** @var string|null */
    private $name;
    public function setName(string $name) : self {
      $this->name = $name;
      return $this;
    }
  }
  $user = new User();
  $user->setName('d');  // << -- error 

```
**Current brhavior:**
Get an error at the last line

**Expect behavior:**
No errors should be reported by plugin

`Report fluent interfaces` option is unchecked.
 
![screenshot from 2019-01-08 10-21-54](https://user-images.githubusercontent.com/425208/50818333-5ad7e600-132f-11e9-9bc9-fdd9ef7ab3c6.png)

**Comment by kalessil** [comment]
I'll check.

**Comment by cosminsandu** [comment]
I also get this error
```php
<?php declare(strict_types=1);

class TEST
{
    private $a;
    private $b;
    private $c;
    public function getA()
    {
        return $this->a;
    }
    public function setA($a): TEST
    {
        $this->a = $a;
        return $this;
    }
    public function getB()
    {
        return $this->b;
    }
    public function setB($b): TEST
    {
        $this->b = $b;
        return $this;
    }
    public function getC()
    {
        return $this->c;
    }
    public function setC($c): TEST
    {
        $this->c = $c;
        return $this;
    }
}

$test = new Test();

$test->setA(1);//Function result is not used
$test->setB(2);//Function result is not used
$test->setC(3);//Function result is not used

// also
$test->setA(1)
    ->setB(2)
    ->setC(3);//Function result is not used
```

**Comment by kalessil** [comment]
Fixed!
